src/class-restapi.php

```diff
@@ -113,22 +113,60 @@ public static function add_results_callback( $data ) {
 			);
 		}
 
+		$env = isset( $parameters['env'] ) ? json_decode( $parameters['env'], true ) : array();
+
+		$php_version = '';
+		if ( isset( $env['php_version'] ) ) {
+			$parts = explode( '.', $env['php_version'] );
+			$php_version = $parts[0] . '.' . $parts[1];
+		}
+
+		$env_name = $env['label'] ?? '';;
+
 		$current_user = wp_get_current_user();
 
-		$args = array(
+		$meta_query = [
+			'relation' => 'AND'
+		];
+
+		if ( $php_version ) {
+			$meta_query[] = array(
+				'key'   => 'php_version',
+				'value' => $php_version,
+			);
+		}
+
+		if ( $env_name ) {
+			$meta_query[] = array(
+				'key'   => 'environment_name',
+				'value' => $env_name,
+			);
+		}
+
+		// Check to see if the test result already exist.
+		$results = get_posts( array(
 			'post_parent' => $parent_id,
 			'post_type'   => 'result',
 			'numberposts' => 1,
 			'author'      => $current_user->ID,
-		);
+			'meta_query'  => $meta_query,
+		) );
 
-		// Check to see if the test result already exist.
-		$results = get_posts( $args );
 		if ( $results ) {
 			$post_id = $results[0]->ID;
 		} else {
-			$results = array(
-				'post_title'   => $current_user->user_login . ' - ' . $slug,
+			$post_title = $current_user->user_login . ' - ' . $slug;
```

Comment:
I think that the `$slug` is a bit redundant here. The user is set as the author, and the changeset is the slug of the parent post.

I guess including the reporter's name makes it easier to know who the result was submitted from. But what about making the title (and thus slug) as just a hash of the changeset number and data sent over? Or the reporter and the hash.

Something like (using a random hash generator):
- `testbot-9ae08922ecb3bf2e199a8fdf769da586`
- `testbot-9ae08922`
- `9ae08922ecb3bf2e199a8fdf769da586` (full hash)
- `9ae08922` (abbreviated hash)

This allows multiple reports for the same commit to exist without the need for `-1` `-2` etc. or adding `php-8-0` or something with the current format. It also accounts for situations where multiple reports are submitted using the same PHP version (maybe different MySQL versions, or different environment types or product offerings).

The test report pages are set to `noindex follow`. So we should not need to be worried about SEO impacts for these. And I don't know that a person is typing in `testbot-r12345` to find specific reports form specific reporters. If the reporter name in the URL is that important, we could register the needed permalink structure/rewrite rules to use the `post_author`.

Reply:
Yes in theory the slug (which is the commit revision number) is a bit redundant. However, it makes for easy lookups.

> It also accounts for situations where multiple reports are submitted using the same PHP version (maybe different MySQL versions, or different environment types or product offerings).

In those cases the reporter does not create new entries, but updates the existing one. AFAIK that's a common scenario and so this behavior must be preserved.

Reply:
Been thinking about this and I don't really feel that strongly about this. But I think we're still going to have some issues if someone submits multiple reports for the same PHP version (maybe with different MySQL versions, or different hosting packages). I'm trying to think of a good way to allow for something like this, but nothing comes to mind other than using some kind of hash of all the environment details passed.

Reply:
> But I think we're still going to have some issues if someone submits multiple reports for the same PHP version (maybe with different MySQL versions, or different hosting packages)

How so?

It's simple: they just need to provide a unique environment name in that case. That sounds like a reasonable ask to me and is part of the setup anyway.

If they don't have unique names and keep overriding their existing entries they will realize that very quickly.